source/ceylon/future/util/sleep.ceylon

```diff
@@ -0,0 +1,18 @@
+import java.lang {
+    Thread,
+    JavaInterruptedException=InterruptedException
+}
+
+"Suspend the current thread for some amount of [[time]]."
+throws (`class InterruptedException`,
+    "If the thread is interrupted before the sleep completes")
+shared void sleep(time) {
+    "The number of milliseconds to sleep for."
+    Integer time;
+    assert (time >= 0);
+    try {
+        Thread.sleep(time);
+    } catch (JavaInterruptedException e) {
```

Comment:
I meant that before throwing the ceylon InterruptedException it should do `Thread.currentThread().interrupt()` so something above will still see the thread as interrupted.